fdbserver/Ratekeeper.actor.cpp

```diff
@@ -917,14 +921,16 @@ Future<Void> refreshStorageServerCommitCost(RatekeeperData* self) {
 			it->value.busiestWriteTagRate = maxRate;
 		}
 
+		auto tempEventCacheHolder = makeReference<EventCacheHolder>(it->key.toString() + "/BusiestWriteTag");
+		self->storageQueueInfoHolder.emplace_back(tempEventCacheHolder);
 		TraceEvent("BusiestWriteTag", it->key)
 		    .detail("Elapsed", elapsed)
 		    .detail("Tag", printable(busiestTag))
 		    .detail("TagOps", maxCost.getOpsSum())
 		    .detail("TagCost", maxCost.getCostSum())
 		    .detail("TotalCost", it->value.totalWriteCosts)
 		    .detail("Reported", it->value.busiestWriteTag.present())
-		    .trackLatest(it->key.toString() + "/BusiestWriteTag");
+		    .trackLatest(tempEventCacheHolder.getPtr()->trackingKey);
```

Comment:
I had not considered `BusiestWriteTag`, but that's a good one to tackle as well. I suppose for this one what we want to do is clear out the existing vector and insert new entries. We can't only append because that will eventually run us out of memory, and if we just maintain a set of all storage servers we've ever seen then it won't erase entries from old storage servers. By clearing and replacing the vector each time, we will get a set of entries that corresponds only with the most recent storage servers that we've interacted with.

Reply:
Sure. I'll use queue to replace the vector. And each time when we run the refreshStorageServerCommitCost function, pop out and destruct all of them and then insert new items. Does it make sense?